pkg/scale/varying_data_type.go

```diff
@@ -58,7 +59,7 @@ type VaryingDataType struct {
 func (vdt *VaryingDataType) Set(value VaryingDataTypeValue) (err error) {
 	_, ok := vdt.cache[value.Index()]
 	if !ok {
-		err = fmt.Errorf("unable to append VaryingDataTypeValue: %T, not in cache", value)
+		err = fmt.Errorf("%w: %T, not in cache", ErrUnableToAddVaryingDataTypeValue, value)
```

Comment:
nit just adding the actual value to the context

```suggestion
		err = fmt.Errorf("%w: %v (%T) is not in cache", ErrUnableToAddVaryingDataTypeValue, value, value)
```